packages/snap-controller/src/Search/SearchController.ts

```diff
@@ -581,3 +698,87 @@ export function generateHrefSelector(element: HTMLElement, href: string, levels
 
 	return;
 }
+
+function getSearchRedirectSchemaData({ redirectURL }: { redirectURL: string }): SearchRedirectSchemaData {
+	return {
+		redirect: redirectURL,
+	};
+}
+
+function getSearchSchemaData({
+	params,
+	store,
+	results,
+}: {
+	params: SearchRequestModel;
+	store: SearchStore;
+	results?: SearchResultStore;
+}): SearchSchemaData {
+	const filters = params.filters?.reduce<{
+		bgfilter?: Array<AutocompleteSchemaDataBgfilterInner>;
+		filter?: Array<AutocompleteSchemaDataFilterInner>;
+	}>((acc, filter) => {
+		const key = filter.background ? 'bgfilter' : 'filter';
+		acc[key] = acc[key] || [];
+
+		const value =
+			filter.type === 'range' &&
+			!isNaN((filter as SearchRequestModelFilterRange).value?.low!) &&
+			!isNaN((filter as SearchRequestModelFilterRange).value?.low!)
+				? [`low=${(filter as SearchRequestModelFilterRange).value?.low}`, `high=${(filter as SearchRequestModelFilterRange).value?.high}`]
+				: [`${(filter as SearchRequestModelFilterValue).value}`];
+
+		const existing = acc[key]!.find((item) => item.field === filter.field);
+		if (existing && !existing.value!.includes(value[0])) {
+			existing.value!.push(...value);
+		} else {
+			acc[key]!.push({
+				field: filter.field,
+				value,
+			});
+		}
+
+		return acc;
+	}, {});
+	return {
+		q: params.search?.query?.string || '',
+		correctedQuery: params.search?.originalQuery,
+		...filters,
+		sort: [
+			{
+				field: store.sorting.current?.field,
+				dir: store.sorting.current?.direction as AutocompleteSchemaDataSortInnerDirEnum,
+			},
+		],
+		pagination: {
+			totalResults: store.pagination.totalResults,
+			page: store.pagination.page,
+			resultsPerPage: store.pagination.pageSize,
+		},
+		merchandising: {
+			personalized: store.merchandising.personalized,
+			redirect: store.merchandising.redirect,
+			triggeredCampaigns:
+				(store.merchandising.campaigns?.length &&
+					store.merchandising.campaigns?.map((campaign) => {
+						const experiement = store.merchandising.experiments.find((experiment) => experiment.campaignId === campaign.id);
+						return {
+							id: campaign.id,
+							experimentId: experiement?.experimentId,
+							variationId: experiement?.variationId,
+						};
+					})) ||
+				undefined,
+		},
+		results:
+			results?.map((result: Product | Banner): Item => {
+				const core = (result as Product).mappings.core!;
+				return {
+					uid: core.uid || '',
+					childUid: core.uid,
+					sku: core.sku,
+					childSku: core.sku,
```

Comment:
Same thing here with regards to the child properties... should we leave them empty or send non-child properties?